### PR TITLE
Handle error if DID chain resolution fails.

### LINKED ISSUE #36: Display "Try again later" on failed HTTP/FFI calls
If the HTTP server is unavailable, the mobile app will throw an exception if an FFI call or HTTP request is made (at least in some cases, such as "show chain" in the credential wallet).

These exceptions should instead be caught, and a message displayed, e.g. "Trustchain server not available. Try again later."

On the device (i.e. when not in debug mode) a spinner is shown indefinitely without crashing the app, so this is an enhancement not a necessity.

### PR DESCRIPTION
**Pull request by pwochner**
Handle different cases for DID chain resolution:
- display message to user in case resolution fails
- show spinning progression indicator if waiting for response.

Closes #36

### CODE REVIEW

**Comment by pwochner** [comment]
Wait until `feat/5-ffi-v1` is merged into `dev`.  Then merge `dev` into `36-error`.

File: lib/app/pages/chain/widget/chain.dart [hunk 1]
```diff
@@ -24,31 +26,24 @@ class DIDChainWidget extends StatelessWidget {
...
+                final item = entry.value;
+                final isFirst = index == 0;
+                final isLast = index == model.data.length - 1;
+
+                return CustomTile(
+                  model: item,
+                  isFirst: isFirst,
+                  isLast: isLast,
+                  rootEventDate:
+                      DateFormat('dd MMM yyyy', 'en_US').format(DateTime.now()),
```
Comment (sgreenbury) [inline lib/app/pages/chain/widget/chain.dart:44]: Should this be 'en_UK'?

**Review by sgreenbury — APPROVED** [review]
Looks great!

File: lib/app/pages/chain/widget/chain.dart (hunk 1 above)
Comment (sgreenbury) [reply lib/app/pages/chain/widget/chain.dart]: Fixed.

**Pull request merged**